src/python/pants/python/python_setup.py

```diff
@@ -127,6 +127,19 @@ def register_options(cls, register):
                 "Mutually exclusive with `[python-setup].requirement_constraints`."
             ),
         )
+        register(
+            "--experimental-resolves-to-lockfiles",
+            advanced=True,
+            type=dict,
+            help=(
+                "A mapping of logical names to lockfile paths used in your project, e.g. "
+                "`{ default = '3rdparty/default_lockfile.txt', py2 = '3rdparty/py2.txt' }`.\n\n"
+                "To generate a lockfile, run `./pants generate-lockfiles --resolve=<name>` or "
+                "`./pants generate-lockfiles` to generate for all resolves (including tool "
+                "lockfiles).\n\n"
```

Comment:
I plan to add more info once targets gain the `resolve` field. So far I have drafted:

```python
"These resolve names allow certain Python targets to choose which lockfile they "
                "should use by setting the field `resolve='<name>'`. If you don't set the field, "
                "they will default to your resolve named `default`.\n\n"
                "Most projects will only need a single resolve for the whole project - this is a "
                "sensible default.\n\n"
```

Reply:
I'm no longer certain we would to force you into setting a default resolve, and for that resolve to be named `default`. Per discussion in https://github.com/pantsbuild/pants/issues/12714, we may want a dedicated option to set the default resolve, if any.

This PR intentionally punts on committing to `default` being a special thing. Here, it's just like any other name.